api/test/teamIntegration.test.ts

```diff
@@ -0,0 +1,1494 @@
+    });
+
+    // Create a test user who is NOT a team member
+    const username = 'outsider';
+    const password = 'password123';
+
+    const [user, err] = await createUser({
+      username,
+      name: 'Outsider User',
+    });
+
+    expect(err).to.equal('');
+    expect(user).to.not.be.null;
+    if (!user) {
+      throw new Error('User creation failed ' + err);
+    }
+
+    // Add basic user permissions
+    addGlobalRole({
+      user,
+      role: Role.GENERAL_CREATOR,
+    });
+
+    await saveCouchUser(user);
+    await addLocalPasswordForUser(user, password);
+
+    // Generate token for this user
+    const signingKey = await KEY_SERVICE.getSigningKey();
+    const upgraded = await upgradeCouchUserToExpressUser({dbUser: user});
+    const userToken = await generateJwtFromUser({user: upgraded, signingKey});
+
+    // Test permissions - this user SHOULD be allowed to create projects globally
+    expect(
+      userCanDo({
+        user: upgraded,
+        action: Action.CREATE_PROJECT,
+        resourceId: undefined,
+      })
+    ).to.be.true;
+
+    // But should NOT be allowed to create projects in a team they don't belong to
+    expect(
+      userCanDo({
+        user: upgraded,
+        action: Action.CREATE_PROJECT_IN_TEAM,
+        resourceId: team._id,
+      })
+    ).to.be.false;
+
+    // Try to create a project in the team using the API - should be denied
+    await requestAuthAndType(
+      request(app)
+        .post(`${NOTEBOOKS_API_BASE}`)
+        .send({
+          name: 'Unauthorized team project',
+          'ui-specification': uispec,
+          metadata: {test_key: 'test_value'},
+          teamId: team._id,
+        }),
+      userToken
+    ).expect(401);
+
+    // Try to create a template in the team using the API - should be denied
+    await requestAuthAndType(
+      request(app)
+        .post(`${TEMPLATE_API_BASE}`)
+        .send({
+          ...getSampleNotebook(),
+          teamId: team._id,
+        }),
+      userToken
+    ).expect(401);
+  });
+
+  it('allows system admin to create resources in any team', async () => {
+    // Create a team
+    const team = await createTeam(app, {
+      teamName: 'Admin Access Team',
+      description: 'A team for admin access tests',
+    });
+
+    // System admin should be able to create projects and templates in any team
+    await requestAuthAndType(
+      request(app)
+        .post(`${NOTEBOOKS_API_BASE}`)
+        .send({
+          name: 'Admin team project',
+          'ui-specification': uispec,
+          metadata: {test_key: 'test_value'},
+          teamId: team._id,
+        }),
+      adminToken
+    ).expect(200);
+
+    await requestAuthAndType(
+      request(app)
+        .post(`${TEMPLATE_API_BASE}`)
+        .send({
+          ...getSampleNotebook(),
+          teamId: team._id,
+        }),
+      adminToken
+    ).expect(200);
+  });
+
+  it('correctly fetches templates by team ID', async () => {
```

Comment:
I get a test failure when running this as part of the overall suite. On it's own it passes. "InternalSystemError: An unexpected error occurred while trying to PUT the new template document into the teams DB. Exception {"status":409,"name":"conflict","message":"Document update conflict"}"